src/main/java/com/woowacamp/soolsool/core/statistics/infra/StatisticsRedisImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.woowacamp.soolsool.core.statistics.infra;
+
+import com.woowacamp.soolsool.core.statistics.domain.StatisticsLiquors;
+import java.util.Optional;
+import org.redisson.api.RMapCache;
+import org.redisson.api.RedissonClient;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StatisticsRedisImpl implements StatisticsRedis {
+
+    private static final String MAP_KEY_STATISTICS = "statistics";
+    private static final String STATISTICS_PRICE = "top5SalePrice";
+    private static final String STATISTICS_QUANTITY = "top5SaleQuantity";
```

Comment:
상수니까 UPPER_CASE로 작성하는게 좋을 것 같아요!

Reply:
오잉?? "statistics" 요 부분이요??

Reply:
```suggestion
    private static final String MAP_KEY_STATISTICS = "STATISTIC";
    private static final String STATISTICS_PRICE = "TOP5_SALE_PRICE";
    private static final String STATISTICS_QUANTITY = "TOP5_SALE_QUANTITY";
```

상수는 UPPER_CASE로 하지 않았나요? 근데 또 다른 레디스 자료를 찾아보니 다음과 같은 패턴을 쓰기도 하더라구요

> 상품 번호가 1이고 총 가격을 나타내는 Key에 대한 네이밍

```
product:1:total:price
```

> 기존(LiquorCtr)에서 사용하던 방식 (1번 상품의 클릭률)

```
LIQUOR_CTR-1
```

@hwihwi523 같이 보시면 좋을 것 같아요~

Reply:
저도 관련해서 stack overflow 나 redis 공식 문서를 찾아봤는데
: 세미콜론 같은 경우 id 처럼 계층 관계가 있는 것인데 해당 부분은 statistics 라는 자료형에 각각 다른 key를 가지고 있어서 네이밍만 수정해도 괜찮을것 같네요^^
추가로 예시보면 보통 소문자로 구성되어있고 경우에 따라 camle case || snake case || hypen 형식으로 다양하게 쓴다고 하는데 이 부분은 다같이 논의해 봐도 좋을것 같습니다^^

https://stackoverflow.com/questions/6965451/redis-key-naming-conventions
https://fjolt.com/article/redis-hashes#:~:text=The%20naming%20convention%20of%20a,value%20key%20value%20....

Reply:
상수는 `STATISTICS`, `TOP5_SALE_PRICE`, `TOP5_SALE_QUANTITY`처럼 UPPER CASE로 하는 게 좋은 것 같습니다!

그리고 저도 세권이형과 마찬가지로 콜론으로 데이터를 다루진 않았었는데요!
그 이유는 계층 관계, 또는 prefix로 설정할 필요가 없었기 때문입니다!
왜냐하면 LIQUOR_CTR 이라는 Map 안에 (Key, Value) 쌍으로 저장하고 있기 때문이었습니다!

e.g. 요것처럼요!
LIQUOR_CTR : {
    1: RedisLiquorCtr(1, 2),
    20: RedisLiquorCtr(412, 313)
}

만약 Map을 쓰지 않았다면 저는 쩡규가 말한 대로 콜론으로 구분할 생각이었습니다!

e.g. 요로코롬요!
LIQUOR_CTR:1 : RedisLiquorCtr(1, 2)
LIQUOR_CTR:20 : RedisLiquorCtr(412, 313)

Reply:
그런데 왜 HashMap과 같은 구조인 Redis hashes를 사용하신건가요? 지금 상황에서는 Redis strings로 해결할 수 있지 않나요?

현재 hashes의 기능을 전혀 사용하지 않는데 메모리를 더 사용하는 hashes를 쓰는 이유가 있나요?

만약 strings로 사용한다면 `liquor_ctr:1`로 사용할 수 있을 것 같아요. 혹은 네이밍이 헷갈리니 `liquor:1:liquor_ctr`로 사용할 수 있을 것 같구요.

Reply:
LIQUOR_CTR hashes 안에서 상품 id와 CTR 정보를 key-value로 사용하고 있습니다.

어느 면에서 hashes의 기능을 전혀 사용하지 않고 있다고 생각하신 걸까요?

Reply:
LIQUOR_CTR 들을 하나의 HashMap에서 모아서 관리하고 있는데 모았다는 특징을 이용하고 있나요? 

그냥 개별의 LIQUOR_CTR들로 사용하면 더 적은 메모리로 똑같은 기능을 구현할 수 있지 않을까요?

Reply:
특정 정보만 범위 지어 관리하는 게.. 조금 걸리신다면.. 말씀대로라면 hashes는 언제 써야 하죠..?

https://mozi.tistory.com/522

그리고 간단히 한 번 참고해 보시면 좋을 것 같아요.

Reply:
hashes가 strings보다 더 효율적이라는 사실을 몰랐습니다... 찾아보고 말할걸 죄송합니다!!!!!!!

Reply:
아닙미다!!! 저도 진짜 어렴풋이만 인지하고 있던 건데 요번 기회에 더 알아보게 되었읍니다!!!!!!! 감삼다!!!!!!!